### DISCUSSION TITLE
Disable 3rd-party JavaScript

### DISCUSSION THREAD

**Discussion by anyone00** [bromite/bromite#2303] [Ideas]
From my experience most third-party JS on website I use is for ads, social media widgets, tracking, telemetry. Disable third-party JS like in ublock origin would be a nice feature to have.

**Comment by uazo** [comment]
interesting idea.
I tried to disable it and actually most of the sites still work, it would be interesting to add an option.
We should think of a way to reactivate the blocked iframe, perhaps by pressing on a notification, to temporarily allow execution until the page is changed, because often videos are embedded in third-party iframes, but we could start with a "simple" new site setting.
it should be understood if it is actually worth it

these are the changes I tried:
```
diff --git a/third_party/blink/renderer/core/frame/local_dom_window.cc b/third_party/blink/renderer/core/frame/local_dom_window.cc
index d8328b501e..fa4a9a9edd 100644
--- a/third_party/blink/renderer/core/frame/local_dom_window.cc
+++ b/third_party/blink/renderer/core/frame/local_dom_window.cc
@@ -388,6 +388,13 @@ bool LocalDOMWindow::CanExecuteScripts(
   if (!GetFrame())
     return false;

+  LOG(INFO) << "---LocalDOMWindow::CanExecuteScripts";
+  if (GetFrame()->IsCrossOriginToOutermostMainFrame()) {
+    LOG(INFO) << "---disabled javascript on " << Url().ElidedString();
+    GetFrame()->GetSettings()->SetScriptEnabled(false);
+    return false;
+  }
+
   // Normally, scripts are not allowed in sandboxed contexts that disallow them.
   // However, there is an exception for cases when the script should bypass the
   // main world's CSP (such as for privileged isolated worlds). See
```

**Reply by ghost** [reply]
I think blocking 3p scripts is less useful without a dedicated menu to manually noop specific sites but as gorhills wiki says 3p iframes breaks a lot less

**Reply by uazo** [reply]
of course, it is possible to provide exceptions via the site settings ui.
the question is whether it is worth it.

>  gorhills wiki says 3p iframes breaks a lot less

can you write the link please?

**Reply by n-buna404** [reply]
https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-Benefits-of-blocking-3rd-party-iframe-tags

**Reply by uazo** [reply]
thank you.
ublock allows fine selection of third party contexts to be unlocked, which is not allowed by any chromium ui. 
we should also think about that (but I already have an idea, perhaps a bit complex)

**Reply by csagan5** [reply]
I doubt that it is the only change needed, but we could try adding this.

**Reply by uazo** [reply]
ok let's try. however, ublock does not disable javascript, but the loading of the iframes at all.

**Reply by uazo** [reply]
So what do you say, considering [this](https://github.com/bromite/bromite/issues/2304#issuecomment-1261438449), we want to try anyway?
